src/librustdoc/theme.rs

```diff
@@ -0,0 +1,379 @@
+            if let Some(ref mut path) = paths.last_mut() {
+                for entry in inner(v, events, pos).iter() {
+                    path.children.insert(entry.clone());
+                }
+            }
+        }
+        if let Some(Events::OutBlock(_)) = get_useful_next(events, pos) {
+            *pos += 1;
+        }
+    }
+    paths.iter().cloned().collect()
+}
+
+pub fn load_css_paths(v: &[u8]) -> CssPath {
+    let events = load_css_events(v);
+    let mut pos = 0;
+
+    let mut parent = CssPath::new("parent".to_owned());
+    parent.children = inner(v, &events, &mut pos);
+    parent
+}
+
+pub fn get_differences(against: &CssPath, other: &CssPath, v: &mut Vec<String>) {
+    if against.name != other.name {
+        return
+    } else {
+        for child in &against.children {
+            let mut found = false;
+            let mut found_working = false;
+            let mut tmp = Vec::new();
+
+            for other_child in &other.children {
+                if child.name == other_child.name {
+                    if child != other_child {
+                        get_differences(child, other_child, &mut tmp);
+                    } else {
+                        found_working = true;
+                    }
+                    found = true;
+                    break
+                }
+            }
+            if found == false {
+                v.push(format!("  Missing \"{}\" rule", child.name));
+            } else if found_working == false {
+                v.extend(tmp.iter().cloned());
+            }
+        }
+    }
+}
+
+pub fn test_theme_against<P: AsRef<Path>>(f: &P, against: &CssPath) -> (bool, Vec<String>) {
+    let mut file = try_something!(File::open(f), (false, Vec::new()));
+    let mut data = Vec::with_capacity(1000);
+
+    try_something!(file.read_to_end(&mut data), (false, Vec::new()));
+    let paths = load_css_paths(&data);
+    let mut ret = Vec::new();
+    get_differences(against, &paths, &mut ret);
+    (true, ret)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_comments_in_rules() {
+        let text = r#"
+rule a {}
+
+rule b, c
+// a line comment
+{}
+
+rule d
+// another line comment
+e {}
+
+rule f/* a multine
+
+comment*/{}
+
+rule g/* another multine
+
+comment*/h
+
+i {}
+
+rule j/*commeeeeent
+
+you like things like "{}" in there? :)
+*/
+end {}"#;
+
+        let against = r#"
+rule a {}
+
+rule b, c {}
+
+rule d e {}
+
+rule f {}
+
+rule gh i {}
+
+rule j end {}
+"#;
+
+        let mut ret = Vec::new();
+        get_differences(&load_css_paths(against.as_bytes()),
+                        &load_css_paths(text.as_bytes()),
+                        &mut ret);
+        assert!(ret.is_empty());
+    }
+
+    #[test]
+    fn test_text() {
+        let text = r#"
+a
+/* sdfs
+*/ b
+c // sdf
+d {}
+"#;
+        let paths = load_css_paths(text.as_bytes());
+        assert!(paths.children.get(&CssPath::new("a b c d".to_owned())).is_some());
```

Comment:
`.get(..).is_some()` is the same as `.contains(..)`

Reply:
Oh indeed, thanks!